contracts/strategies/1inch/base/OneInchStrategy_DAI_X.sol

```diff
@@ -0,0 +1,362 @@
+  * and converts accordingly for reinvesting underlying
+  */
+  function liquidateReward() internal {
+    if (!sell()) {
+      // Profits can be disabled for possible simplified and rapid exit
+      emit ProfitsNotCollected(sell(), false);
+      return;
+    }
+
+    rewardTokensToCommon();
+
+    uint256 rewardBalance = IERC20(weth).balanceOf(address(this));
+
+    // share 30% of the wrapped Ether as a profit sharing reward
+    notifyProfitInRewardToken(rewardBalance);
+    uint256 remainingRewardBalance = IERC20(rewardToken()).balanceOf(address(this));
+
+    if (remainingRewardBalance == 0) {
+      return;
+    }
+
+    commonToDepositTokens(remainingRewardBalance);
+
+    depositMooniSwap();
+  }
+
+  /**
+  * swaps the common token (WETH) to the deposit tokens for reinvesting
+  */
+  function commonToDepositTokens(uint256 amount) internal {
+    // convert half of the remaining rewardBalance back to the main deposit token DAI
+    uint256 liquidateRewardToDaiBalance = amount.div(2);
+
+    if(liquidateDepositTokenViaDai()) {
+      // if we liquidate the deposit token via DAI then we swap all of the rewardBalance to DAI
+      liquidateRewardToDaiBalance = amount;
+    }
+
+    swapViaULorOneInchUL(weth, dai, liquidateRewardToDaiBalance);
+
+    if(liquidateDepositTokenViaDai()) {
+      // convert half of the dai balance to the second token
+      uint256 remainingRewardBalance = IERC20(dai).balanceOf(address(this));
+      swapViaULorOneInchUL(dai, depositToken(), remainingRewardBalance);
+    } else {
+      // the remaining half of the reward balance is converted into the second token
+      uint256 remainingRewardBalance = IERC20(rewardToken()).balanceOf(address(this));
+      swapViaULorOneInchUL(weth, depositToken(), remainingRewardBalance);
+    }
+  }
+
+  /**
+  * swaps all the reward tokens to WETH
+  */
+  function rewardTokensToCommon() internal {
+    // multiple reward tokens are supported -> liquidate all of them into common rewardToken (weth)
+    for(uint256 i = 0; i < rewardTokens.length; i++){
+      address rewardToken = rewardTokens[i];
+      uint256 rewardBalance = IERC20(rewardToken).balanceOf(address(this));
+      if (rewardBalance == 0 || (storedLiquidationDexes[rewardToken][weth].length < 1 && storedLiquidationDexes[rewardToken][eth].length < 1)) {
+        continue;
+      }
+
+      swapViaULorOneInchUL(rewardToken, weth, rewardBalance);
+    }
+  }
+
+  function swapViaULorOneInchUL(address fromToken, address toToken, uint256 amount) internal {
+      bool toWethHasToBeWrapped = false;
+      if(toToken == weth) {
+        // check if a path for weth exists or if we have to wrap
+        toWethHasToBeWrapped = storedLiquidationDexes[fromToken][toToken].length < 1;
+      }
+      bytes32[] memory dexes = storedLiquidationDexes[rewardToken()][weth];
+      if(toWethHasToBeWrapped) {
+        dexes = storedLiquidationDexes[rewardToken()][eth];
+      }
+
+      if(dexes[0] == oneInchDex) {
```

Comment:
This is not needed! The swap can go though the `universalLiquidator`, just with the 1INCH byte identifier in `storedLiquidationDexes`